src/common/otp.c

```diff
@@ -13,3 +13,16 @@ const char *otp_get_mac_address_str() {
     }
     return mac_address_str;
 }
+
+const char *otp_get_serial_number() {
+    static char serial_number_str[16] = { 0 };
```

Comment:
Shouldn't you make the size be `OTP_SERIAL_NUMBER_SIZE`? You have a mismatch in the size of the array and the upper bound of the `i` when you write to it below. I assume they're equal and chances of `OTP_SERIAL_NUMBER_SIZE` changing are low, but anyway.